merkletree/doc.go

```diff
@@ -0,0 +1,25 @@
+/*
+Package merkletree implements a Merkle prefix tree and related data structures.
+The Merkle prefix tree is one of the most important components of the CONIKS protocol.
+We implemented this data structure separately as a library
+to help other developers use it in their implementation easily.
+
+Persistent Authenticated Dictionary
+
+This module implements a persistent authenticated dictionary (PAD) data structure.
+
+Merkle Prefix Tree
+
+This module implements the Merkle prefix tree, which is the data structure
+underlying our PAD implementation. It is a binary tree, and that there are
+two types of leaf nodes: empty leaf node and user leaf node. Each node contains
+its prefix index and its level value. It provides methods to insert, update
+(to update an existing key-value pair) and lookup. The tree is append-only,
```

Comment:
You can update the value in the tree and there's no record of its previous content.  It alone isn't, as I understand it.  The PAD, however, is only updated once an epoch and contains a snapshot of previous values.